thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/cube/data/node/CubeNodeUtils.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.thirdeye.cube.data.node;
+
+import java.util.List;
+import org.apache.commons.lang.ObjectUtils;
+
+
+public class CubeNodeUtils {
+
+  /**
+   * Check if the hierarchical tree of the given two root nodes are the same.
+   *
+   * @param node1 the root node of the first hierarchical tree.
+   * @param node2 the root node of the second hierarchical tree.
+   *
+   * @return true if both hierarchical tree are the same.
+   */
+  public static boolean equalHierarchy(CubeNode node1, CubeNode node2) {
+    return equalHierarchy(node1, null, node2, null);
+  }
+
+  public static boolean equalHierarchy(CubeNode node1, CubeNode node1Parent, CubeNode node2, CubeNode node2Parent) {
+    boolean sameData = ObjectUtils.equals(node1, node2);
+    if (sameData) {
```

Comment:
It would be more clear if using:
if (!sameData) return false;
// other logics  .....

Reply:
Awesome! Will flip it when I am working on the Ratio Cube code.

Reply:
Addressed in the PR for ratio cube algorithm: https://github.com/apache/incubator-pinot/pull/4246